WooCommerce/Classes/ViewRelated/Orders/Order Creation/EditableOrderViewModel.swift

```diff
@@ -903,11 +903,9 @@ private extension EditableOrderViewModel {
 
     /// Adds a selected product (from the product list) to the order.
     ///
-    func addOrRemoveProductToOrder(_ product: Product) {
+    func changeSelectionStateForProduct(_ product: Product) {
         // Needed because `allProducts` is only updated at start, so product from new pages are not synced.
-        if !allProducts.contains(product) {
-            allProducts.append(product)
-        }
```

Comment:
Question: Will not be needing this when dealing with pagination? Perhaps not anymore since this was added when the loading was lazy? 🤔

Reply:
With set we're basically doing the same as before: Insert the given element in the set if it is not already present.